astropy/io/fits/connect.py

```diff
@@ -392,3 +446,170 @@ def write_table_fits(input, output, overwrite=False):
+    ``(format, width, precision, exponential)`` tuple (the TDISP values
+    for ASCII and binary are identical except for 'Lw',
+    which is only present in BINTABLE extensions
+
+    Parameters
+    ----------
+    tdisp: str
+        TDISPn FITS Header keyword.  Used to specify display formatting.
+
+    Returns
+    -------
+    formatc: str
+        The format characters from TDISPn
+
+    width: str
+        The width int value from TDISPn
+
+    precision: str
+        The precision int value from TIDISPn
+
+    exponential: str
+        The exponential int value from TDISPn
+
+    """
+
+    # Use appropriate regex for format type
+    tdisp = tdisp.strip()
+    fmt_key = tdisp[0] if tdisp[0] !='E' or tdisp[1] not in 'NS' else tdisp[:2]
+    try:
+        tdisp_re = TDISP_RE_DICT[fmt_key]
+    except KeyError:
+        raise VerifyError('Format {} is not recognized.'.format(tdisp))
+
+
+    match = tdisp_re.match(tdisp.strip())
+    if not match or match.group('formatc') is None:
+        raise VerifyError('Format {} is not recognized.'.format(tdisp))
+
+    formatc = match.group('formatc')
+    width = match.group('width')
+    precision = None
+    exponential = None
+
+    # Some formats have
+    if tdisp[0] in ('I', 'B', 'O', 'Z', 'F', 'E', 'G', 'D'):
+        precision = match.group('precision')
+        if precision is None:
+            precision = 1
+    if tdisp[0] in ('E', 'D', 'G') and tdisp[1] not in ('N', 'S'):
+        exponential = match.group('exponential')
+        if exponential is None:
+            exponential = 1
+
+    # Once parsed, check format dict to do conversion to a formatting string
+    return formatc, width, precision, exponential
+
+
+def _fortran_to_python_format(tdisp):
+    """
+    Turn the TDISPn fortran format pieces into a final Python format string.
+    See the format_type definitions above the TDISP_FMT_DICT. If codes is
+    changed to take advantage of the exponential specification, will need to
+    add it as another input parameter.
+
+    Parameters
+    ----------
+    tdisp: str
+        TDISPn FITS Header keyword.  Used to specify display formatting.
+
+    Returns
+    -------
+    format_string: str
+        The TDISPn keyword string translated into a Python format string.
+    """
+    format_type, width, precision, exponential = _parse_tdisp_format(tdisp)
+
+    try:
+        fmt = TDISP_FMT_DICT[format_type]
+        return fmt.format(width=width, precision=precision)
+
+    except KeyError:
+        raise VerifyError('Format {} is not recognized.'.format(format_type))
+
+
+def python_to_tdisp(format_string, logical_dtype = False):
+    """
+    Turn the Python format string to a TDISP FITS compliant format string. Not
+    all formats convert. these will cause a Warning and return None.
+
+    Parameters
+    ----------
+    format_string: str
+        TDISPn FITS Header keyword.  Used to specify display formatting.
+
+    logical_dtype: bool
```

Comment:
The logical format is a very specific exception to this situation, as it will round trip to a character format (since there's no logical Python format type).  I feel like it would be more awkward to have this edited in the call site. Or rather it makes the most sense to me to have python_to_tdisp be a fully cohesive translation with any special cases handled inside of it.  And if it's any other dtype we don't really need to know, the function will handle it correctly.  Thoughts?

Reply:
Yes, even when I wrote the comment I wasn't quite sure. I do see your point about keeping things together. How about passing in the actual dtype instead of a bool that has to be constructed by the call site as well?